src/Nethermind/Nethermind.Serialization.Json/UInt256Converter.cs

```diff
@@ -29,8 +29,18 @@ private static UInt256 ReadInternal(ref Utf8JsonReader reader, JsonTokenType all
     {
         if (reader.TokenType == JsonTokenType.Number)
         {
-            return reader.GetUInt64();
+            if (reader.TryGetUInt64(out ulong smallValue))
```

Comment:
If this fails, and the request was chunked, TryGetUint64 will materialize the sequence and then it will be materialized one more time below. I'd either add a length condition here (the lenght of encoded uint64 max value) to not parse through or materialize the `bytes` first and then try parse first Uint64 and then Uint256 if this fails.

Reply:
Length is not a good test, because decimal numbers of the same length can be more or less than max long. But as for reallocating, looks good after extracting it from standard methods, adding it